Bundle/WidgetMapBundle/Entity/WidgetMap.php

```diff
@@ -0,0 +1,451 @@
+     *
+     * @ORM\Column(name="position", type="string", nullable=true)
+     */
+    protected $position;
+
+    /**
+     * @var Collection
+     * @ORM\OneToMany(targetEntity="\Victoire\Bundle\WidgetMapBundle\Entity\WidgetMap", mappedBy="parent")
+     */
+    protected $children;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(name="slot", type="string", length=255, nullable=true)
+     */
+    protected $slot;
+
+    public function __construct()
+    {
+        $this->children = new ArrayCollection();
+        $this->substitutes = new ArrayCollection();
+    }
+
+    /**
+     * @return int
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    public function setId($id)
+    {
+        $this->id = $id;
+    }
+
+    /**
+     * @return null
+     */
+    public function isAsynchronous()
+    {
+        return $this->asynchronous;
+    }
+
+    /**
+     * @param null $asynchronous
+     */
+    public function setAsynchronous($asynchronous)
+    {
+        $this->asynchronous = $asynchronous;
+    }
+
+    /**
+     * Set the action.
+     *
+     * @param string $action
+     *
+     * @throws \Exception The action is not valid
+     */
+    public function setAction($action)
+    {
+        //test validity of the action
+        if ($action !== self::ACTION_CREATE && $action !== self::ACTION_OVERWRITE && $action !== self::ACTION_DELETE) {
+            throw new \Exception('The action of the widget map is not valid. Action: ['.$action.']');
+        }
+
+        $this->action = $action;
+    }
+
+    /**
+     * Get the action.
+     *
+     * @return string The action
+     */
+    public function getAction()
+    {
+        return $this->action;
+    }
+
+    /**
+     * @return Widget
+     */
+    public function getWidget()
+    {
+        return $this->widget;
+    }
+
+    /**
+     * @param Widget $widget
+     *
+     * @return $this
+     */
+    public function setWidget(Widget $widget)
+    {
+        $this->widget = $widget;
+
+        return $this;
+    }
+
+    /**
+     * @return View
+     */
+    public function getView()
+    {
+        return $this->view;
+    }
+
+    /**
+     * @param View $view
+     */
+    public function setView(View $view)
+    {
+        $this->view = $view;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getReplaced()
+    {
+        return $this->replaced;
+    }
+
+    /**
+     * @param mixed $replaced
+     */
+    public function setReplaced($replaced)
+    {
+        if ($replaced) {
+            $replaced->addSubstitute($this);
+        }
+        $this->replaced = $replaced;
+    }
+
+    /**
+     * @return string
+     */
+    public function getSlot()
+    {
+        return $this->slot;
+    }
+
+    /**
+     * @param string $slot
+     */
+    public function setSlot($slot)
+    {
+        $this->slot = $slot;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getChildren(View $view = null)
```

Comment:
please refactor and add phpdoc for :

- [ ] getChildren
- [ ] getChild
- [ ] getChilds